lib/Sema/TypeCheckProtocolInference.cpp

```diff
@@ -782,90 +782,59 @@ Type AssociatedTypeInference::computeFixedTypeWitness(
                                             AssociatedTypeDecl *assocType) {
   // Look at all of the inherited protocols to determine whether they
   // require a fixed type for this associated type.
-  Type dependentType = assocType->getDeclaredInterfaceType();
   Type resultType;
   for (auto conformedProto : adoptee->getAnyNominal()->getAllProtocols()) {
     if (!conformedProto->inheritsFrom(assocType->getProtocol()))
       continue;
 
-    auto genericSig = conformedProto->getGenericSignature();
-    if (!genericSig) return Type();
+    if (!conformedProto->getGenericSignature()) {
+      return Type();
+    }
+
+    const auto contextSelf = cast<ArchetypeType>(
+        conformedProto
+            ->mapTypeIntoContext(conformedProto->getSelfInterfaceType())
+            .getPointer());
 
-    Type concreteType = genericSig->getConcreteType(dependentType);
-    if (!concreteType) continue;
+    const auto nestedType =
+        contextSelf->getNestedType(assocType->getName())->mapTypeOutOfContext();
+
+    if (nestedType->isEqual(conformedProto->getSelfInterfaceType())) {
```

Comment:
A nested type of an archetype will never equal a generic parameter - you’d have to compare against the Self archetype here instead

Reply:
It works though. Have you noticed I'm mapping the nested type out of context?